### PR TITLE
Move weight calculations to hook

### PR DESCRIPTION
**Pull request by sspoon865**

### CODE REVIEW

File: src/lib/hooks/laneRouteHooks.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
+import { atom, useAtomValue } from "jotai";
+
 type TicketType = "Normal" | "PendingInbound" | "PendingMultiProduct";
 
 interface LaneFormState {
 	readyToSubmit: boolean;
 	ticketType: TicketType;
 }
 
+export const liveWeightAtom = atom<number>(0);
```
Comment (ericwilhite) [inline src/lib/hooks/laneRouteHooks.ts:10]: bye bye

File: src/lib/hooks/laneRouteHooks.ts [hunk 2]
```diff
@@ -21,7 +23,7 @@ export function useScaleOrManualWeights(args: {
 	const acceptedUnits = args.acceptedUnits;
 	const laneFormState = args.laneFormState ?? { readyToSubmit: false, ticketType: "Normal" };
 	// If a scale UUID is passed, try to get the live weight. Default to 0 if no scale is found.
-	const liveWeight = args.liveWeight ?? 0;
+	const liveWeight = useAtomValue(liveWeightAtom);
 	const selectedInboundPendingTicket = args.selectedInboundPendingTicket;
```
Comment (ericwilhite) [inline src/lib/hooks/laneRouteHooks.ts:27]: just say no to destructuring

File: src/routes/LaneRoute.tsx [hunk 3]
```diff
@@ -48,13 +49,13 @@ import { twMerge } from "tailwind-merge";
...
 import { UDF } from "@/components/UDF";
-import { atom, useAtom } from "jotai";
+import { atom, useAtom, useSetAtom } from "jotai";
 import { isRightSidebarOpenAtom } from "@/lib/globals";
 import { toast } from "react-hot-toast";
 import { addDays, differenceInDays, isSameYear } from "date-fns";
 import { UseQueryResult } from "@tanstack/react-query";
 import { useCashCalculation } from "@/lib/hooks/useCashCalculation";
-import { useScaleOrManualWeights } from "@/lib/hooks/laneRouteHooks";
+import { liveWeightAtom, useScaleOrManualWeights } from "@/lib/hooks/laneRouteHooks";
```
Comment (ericwilhite) [inline src/routes/LaneRoute.tsx:58]: bye?

**Review by ericwilhite — CHANGES_REQUESTED** [review]

**Comment by sspoon865** [comment]
@ericwilhite Okay this time, it should be a legit re-request for review. 😂 IDK why it kept pinging for the drafts, sorry about that.

**Review by ericwilhite — APPROVED** [review]

**Pull request merged**